src/ChecksumConverter.cc

```diff
@@ -45,7 +45,7 @@
 #endif
 #include <ctype.h>
 
-#if defined(vxWorks) || defined(_WIN32) || defined(__rtems__)
+#if (defined(vxWorks) || defined(_WIN32) || defined(__rtems__)) && !defined(__MINGW32__)
 // These systems have no strncasecmp
 static int strncasecmp(const char *s1, const char *s2, size_t n)
 {
```

Comment:
Does MINGW32 define _WIN32  too? Does it define _MSC_VER?
Maybe this is not needed for Windows any longer. It seems Windows has _strnicmp() now.
I need to check  newer versions of rtems and vxWorks too.

Reply:
_WIN32 is defined for all compilers targeting Windows. _MSC_VER is specific to MSVC compiler.

Reply:
In that case, I plan the following:
```
#ifdef _MSC_VER
#define strncasecmp _strnicmp
#endif

#if defined(vxWorks) || defined(__rtems__)
// These systems have no strncasecmp at the moment
// But avoid compiler errors in case strncasecmp exists in future versions
static int mystrncasecmp(const char *s1, const char *s2, size_t n)
{
    int r=0;
    while (n && (r = toupper(*s1)-toupper(*s2)) == 0) { n--; s1++; s2++; };
    return r;
}
#define strncasecmp mystrncasecmp
#endif
```

Reply:
I will update our build with this and see if we notice any issues. Thanks @xiaoqiangwang for the clarification.

Reply:
Pushed https://github.com/paulscherrerinstitute/StreamDevice/pull/82/commits/3015430e3d883c332ae76e7a2c55a44d328a3a08 with Dirk's changes.